python/tests/expressions/test_expressions_base.py

```diff
@@ -1368,6 +1381,841 @@ def test_bound_boolean_expression_visitor_raise_on_unbound_predicate():
     assert "Not a bound predicate" in str(exc_info.value)
 
 
+def _to_byte_buffer(field_type: IcebergType, val: Any):
```

Comment:
`base` is getting a bit big. Can we refactor it soon?
* `__init__.py` with the unbound expression classes
* `visitors.py` with visitor methods and base classes
* `bound.py` with bound expression classes (and maybe the binding-related visitors?)

Reply:
Yes, I like that. Splitting out `base.py` is on my list, but didn't want to do that in a PR that introduces new functionality.